### PR TITLE
[MB-10099] Validate facility info exists on moves with NTS-release shipment on service counselor submission

### PR DESCRIPTION
**Pull request by hannaliebl**
## Description

Add a check to `UpdateStatusServiceCounselingCompleted` method in move task order updater that returns an error if the move contains an NTS-release shipment that doesn't have facility info.

## Reviewer Notes

- Is this the best place to do this check?
- I am passing nils in for the error and validation error in the error creator because I don't create them - is there a better way to do this?

## Setup

```sh
make server_run
make client_run
```
I took a move in the dev DB that had the correct NTS-release type and just manually changed its status to NEEDS SERVICE COUNSELING, then you can do a PATCH in postman to: `http://officelocal:3000/ghc/v1/move-task-orders/3733132b-f822-43d5-b6b6-a468bfec971e/status/service-counseling-completed` to see the error message.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-10099) for this change

## Screenshots

JSON returned from 422 response in postman:

Screen Shot 2021-11-15 at 2 18 09 PM

### CODE REVIEW

**Review by jenniferlynparsons — COMMENTED** [review]
This looks good to me, but I'd like to see more experienced eyes review it before hitting the approve button.

File: pkg/services/move_task_order/move_task_order_updater_test.go [hunk 1]
```diff
@@ -49,6 +49,26 @@ func (suite *MoveTaskOrderServiceSuite) TestMoveTaskOrderUpdater_UpdateStatusSer
...
+			MTOShipment: models.MTOShipment{
+				ShipmentType: models.MTOShipmentTypeHHGOutOfNTSDom,
+			},
+			Move: noFacilityInfoMove,
+		})
+		eTag := etag.GenerateEtag(noFacilityInfoMove.UpdatedAt)
+
+		_, err := mtoUpdater.UpdateStatusServiceCounselingCompleted(suite.AppContextForTest(), noFacilityInfoMove.ID, eTag)
+
+		suite.IsType(apperror.InvalidInputError{}, err)
```
Comment (reggieriser) [inline pkg/services/move_task_order/move_task_order_updater_test.go:69]: Nit: We may want to check the error message -- or at least some part of it -- to make sure we're getting the specific error so we don't accidentally pass a different `InvalidInputError` (or `ConflictError` if you decide to change to that).

File: pkg/services/move_task_order/move_task_order_updater.go [hunk 2]
```diff
@@ -59,6 +59,13 @@ func (o moveTaskOrderUpdater) UpdateStatusServiceCounselingCompleted(appCtx appc
 		return &models.Move{}, apperror.NewConflictError(move.ID, err.Error())
 	}
 
+	for _, s := range move.MTOShipments {
+		if s.ShipmentType == models.MTOShipmentTypeHHGOutOfNTSDom && s.StorageFacility == nil {
```
Comment (reggieriser) [inline pkg/services/move_task_order/move_task_order_updater.go:63]: I ran it using your instructions and got the error message as expected.  I then added a storage facility and address and tied it to the shipment in the DB but running it again still failed with the same error.  I think the problem may be that you're looking at `s.StorageFacility` but that is not eagerly loaded in the `FetchMoveTaskOrder`.  I don't think you need to eagerly load it necessarily, but instead you can just look at `s.StorageFacilityID` and make sure that's not nil.

**Review by reggieriser — CHANGES_REQUESTED** [review]
The placement of the code and the general logic looks fine.  I'm requesting changes though because I don't think any NTS-R move will pass right now unless I messed up in testing (details in my inline comment).  You may want to add a test that adds a facility/address to an NTS-R move and makes sure it succeeds (pull in master if you do so you have access to `MakeStorageFacility` in testdatagen that recently got merged).

There are a few other small things I noted for your consideration.

**Comment by hannaliebl** [comment]
@reggieriser thank you! I've updated this for you to look at again.

**Review by reggieriser — APPROVED** [review]
Looks good!  I tested it out again and this time was able to get an NTS-R shipment with a storage facility through fine.  Still see the correct error message if there's no facility.  🚢

**Pull request merged**